sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -214,8 +214,8 @@ namespace Azure { namespace Core {
     }
 
     /**
-     * @brief Get the application context (root).
+     * @brief The application context (root).
      */
-    static Context& GetApplicationContext();
+    static AZ_CORE_DLLEXPORT Context ApplicationContext;
```

Comment:
As I've said before, I'm perfectly happy with using static storage duration variables with dynamic initialization (like this one) if we document one of:
* the set of APIs that customers must not call during static initialization to avoid undefined behavior, or
* the set of APIs that customers may call during static initialization with guaranteed undefined behavior, or
* that customers are forbidden to interact with the library during static initialization and must wait for `main` to begin execution.

Absent that documentation, we're likely to get bug reports when customers do what customers do and take dependencies on whatever compiles (today) and passes tests, only to break when we release a new version of the SDK. Or even worse, it might break when a customer updates some unrelated third-party dependency which happens to result in a change in how the linker orders TUs in the exe.

Reply:
Thank you Casey! @RickWinter, I'd say we update our docs to say that customers should not interact with our SDK before `main()`.

Reply:
@JeffreyRichter, how do you feel about telling customers to not interact with our SDK before `main()`?

Reply:
I'm fine with Casey's guidance.